utils/obsproc/NetCDFToIodaConverter.h

```diff
@@ -93,6 +93,52 @@ namespace gdasapp {
       location += other.location;
       oops::Log::trace() << "IodaVars::IodaVars done appending." << std::endl;
     }
+    void trim(const Eigen::Array<bool, Eigen::Dynamic, 1>& mask ) {
+      int newlocation = mask.count();
+
+      Eigen::ArrayXf longitudeMasked(newlocation);
+      Eigen::ArrayXf latitudeMasked(newlocation);
+      Eigen::Array<int64_t, Eigen::Dynamic, 1> datetimeMasked(newlocation);
+      Eigen::ArrayXf obsValMasked(newlocation);
+      Eigen::ArrayXf obsErrorMasked(newlocation);
+      Eigen::ArrayXi preQcMasked(newlocation);
+
+      Eigen::ArrayXXf floatMetadataMasked(newlocation, nfMetadata);
+      Eigen::ArrayXXf intMetadataMasked(newlocation, niMetadata);
+
+      // this feels downright crude, but apparently numpy-type masks are on the todo list for Eigen
+      int j = 0;
+      for (int i = 0; i < location; i++) {
+        if (mask(i)) {
+          longitudeMasked(j) = longitude(i);
+          latitudeMasked(j) = latitude(i);
+          obsValMasked(j) = obsVal(i);
+          obsErrorMasked(j) = obsError(i);
+          preQcMasked(j) = preQc(i);
+          datetimeMasked(j) = datetime(i);
+          for (int k = 0; k < nfMetadata; k++) {
+            intMetadataMasked(j, k) = intMetadata(i, k);
+            }
+          for (int k = 0; k < niMetadata; k++) {
+            intMetadataMasked(j, k) = intMetadata(i, k);
+            }
+        j++;
+        }  // end if (mask(i))
+      }
+
+      longitude = longitudeMasked;
+      latitude = latitudeMasked;
+      datetime = datetimeMasked;
+      obsVal = obsValMasked;
+      obsError = obsErrorMasked;
+      preQc = preQcMasked;
+      floatMetadata = floatMetadataMasked;
+      intMetadata = intMetadataMasked;
+
```

Comment:
I'm actually not convinced this is going to work @AndrewEichmann-NOAA . Does the ```=``` above properly resizes the eigen arrays? Did you check the size of the iodafile that was created?

Reply:
@guillaumevernieres I believe this will work fine. `=` will replace the existing arrays with the new ones which are of the masked size. I'm not sure it would make sense to explicitly do a std::move... I'm guessing the compiler will do that for you anyways when you do an optimized build...

Reply:
(I responded this morning, but for some reason github is treating the comment like a code review that needs to be submitted??)

Reply:
@AndrewEichmann-NOAA All std::move does is steal the pointer...  You can avoid unnecessary copies if you use it correctly...

Reply:
Now that I'm using a whole class instance, I wonder if I can do it to `this` - feels a little Indian rope trick-like

Reply:
sounds good to me @AndrewEichmann-NOAA , we'll switch to `_` in a later PR!